pkg/kyc/component.go

```diff
@@ -663,3 +682,27 @@ func (c *component) sendSmsCodeGeneric(ctx context.Context, accessToken string,
 
 	return *smsCodeKc.Code, err
 }
+
+func (c *component) zipAttachments(attachments []apikyc.AttachmentRepresentation) ([]byte, error) {
```

Comment:
I would create an external tool in /internal/xxx responsible of ZIP creation.
This way, you will be able to have unit tests dedicated to the zip creation... and avoid having to mock-up zip processing here.
(ex: call a NewZipWriter in NewComponent and override the value in unit tests)

Reply:
Why not, but we actually do not need to mock anything of this part to test it

Reply:
Of course we don't need to mock anything... there is no unit test covering this part.

Reply:
I did a dedicated test : TestZipAttachments

Reply:
Maybe Francis was thinking about a test of the public method with multiple attachments. For now, we have a test which covers the private method in charge of the ZIP but not a one which ensure a ZIP is created if you provide multiple attachments.